### PR TITLE
[magnum/magnum-plugins] Fix patches for --head builds

### PR DESCRIPTION
**Pull request by Squareys**
Hi all!

I updated the magnum and magnum-plugins portfiles to work with current HEAD.
As per usual the maintainer of Magnum, @mosra, will quickly review and approve the PR.

Best,
Jonathan

### CODE REVIEW

**Review by mosra — APPROVED** [review]

File: ports/magnum/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,25 @@
 include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/magnum/portfile.cmake]: Can you remove this line? We don't need to include this now.

File: ports/magnum/portfile.cmake [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+
+# Patches that are independent of --head flag
+set(_PATCHES 002-tools-path.patch)
```
Comment (JackBoosY) [inline ports/magnum/portfile.cmake]: Shouldn't this patch be after 001?

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please update version info of these ports, see [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

File: ports/magnum/portfile.cmake (hunk 2 above)
Comment (Squareys) [reply ports/magnum/portfile.cmake]: In this particular case the order doesn't matter, but I'll rename the files to have them reflect the current order

**Comment by Squareys** [comment]
@JackBoosY Thank you for the review!

I made changes accordingly. Please have another look.

**Comment by JackBoosY** [comment]
@Squareys Don't forget to update version info : )
[Documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Comment by Squareys** [comment]
@JackBoosY Ah, right, thanks! 🙏 

Done with latest commit.

**Comment by JackBoosY** [comment]
And `magnum-plugins` version update?

**Comment by Squareys** [comment]
Yes, sorry, @mosra just pinged me on that, too. Totally forgot I had changed that aswell 🙄

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM, we need to test all features.

**Comment by JackBoosY** [comment]
Build failure with `magnum[al-info,anyimageimporter,anyaudioimporter,anyimageconverter,anysceneimporter,audio,debugtools,distancefieldconverter,fontconverter,gl,gl-info,glfwapplication,imageconverter,magnumfont,magnumfontconverter,meshtools,objimporter,tgaimageconverter,opengltester,primitives,sdl2application,scenegraph,shaders,text,texturetools,tgaimporter,trade,wavaudioimporter,windowlesswglapplication,eglcontext,cglcontext,glxcontext,wglcontext,windowlesseglapplication,eglcontext,windowlessglxapplication]:x86-windows`:

```
CMake Error at src/Magnum/Platform/CMakeLists.txt:515 (message):
  X11 library, required by some applications, was not found.  Set
  WITH_*X*APPLICATION to OFF to skip building them.
```

Looks like you forgot to pass in the following parameters:
```
CORRADE_TARGET_UNIX          - Defined if compiled for some Unix flavor
CORRADE_TARGET_APPLE         - Defined if compiled for Apple platforms
CORRADE_TARGET_IOS           - Defined if compiled for iOS (device or
CORRADE_TARGET_IOS_SIMULATOR - Defined if compiled for iOS Simulator
CORRADE_TARGET_WINDOWS       - Defined if compiled for Windows
CORRADE_TARGET_WINDOWS_RT    - Defined if compiled for Windows RT
CORRADE_TARGET_EMSCRIPTEN    - Defined if compiled for Emscripten
CORRADE_TARGET_ANDROID       - Defined if compiled for Android
```

**Comment by Squareys** [comment]
@JackBoosY The magnum port has platform dependent features, e.g. `windowlessglxapplication`. If you build `[*]`, then all are included. We have an open PR on that https://github.com/mosra/magnum/pull/368, but didn't come to a consensus on how this should behave.

Do you have a suggestion?

CC @mosra

**Comment by mosra** [comment]
The inability to use `[*]` is a separate problem that needs a dedicated solution from our side and due to time constraints we weren't able to resolve that yet. @JackBoosY Would it be possible to merge this as-is? 

The build should work on Windows if you install `magnum[al-info,anyimageimporter,anyaudioimporter,anyimageconverter,anysceneimporter,audio,debugtools,distancefieldconverter,fontconverter,gl,gl-info,glfwapplication,imageconverter,magnumfont,magnumfontconverter,meshtools,objimporter,tgaimageconverter,opengltester,primitives,sdl2application,scenegraph,shaders,text,texturetools,tgaimporter,trade,wavaudioimporter,windowlesswglapplication,wglcontext]` (i.e., with `eglcontext`, `cglcontext`, `glxcontext`, `windowlesscglapplication`, `windowlesseglapplication`, `windowlessglxapplication`, which are Linux- and macOS-specific, omitted).

**Comment by JackBoosY** [comment]
@Squareys If some features are not supported by some triplets, you should add error message in _portfile.cmake_ before `vcpkg_configure_cmake`.
Such as:
```
if ("eglcontext" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_WINDOWS)
    message(FATAL_ERROR "Feature eglcontext only support windows")
endif()
```

**Comment by mosra** [comment]
Okay, we'll need to roll out our own `FindOpenAL` that correctly links to `winmm` on Windows (or uses the exported OpenAL Soft CMake config), the builtin CMake module doesn't. Is that a blocker for this PR? This issue was there probably since ever.

Good idea with the `FATAL_ERROR`.

**Comment by JackBoosY** [comment]
@mosra Yes, we need all features to be successfully tested.

**Comment by dan-shaw** [comment]
@mosra @Squareys @JackBoosY It is okay if not all features are passing. We just don't want to introduce any additional failures (which is why we test all the features). Since this feature combination seems to have been failing since the beginning, I would be OK with merging this PR.

**Review by dan-shaw — APPROVED** [review]

**Comment by JackBoosY** [comment]
@dan-shaw Okay.

**Pull request merged**